### ISSUE TITLE
Kafka consumer stop consume messsage (Kafka consumer does not send “TryReceiveMessage” to Event Hub)

### ISSUE THREAD

**Issue by huuthanghust** [Azure/azure-event-hubs-for-kafka#72]
Description
===========
Kafka consumer stop consume messsages randomly. If I restart service and then it start to assyned partition again.

How to reproduce
================
cloned azure-event-hubs-for-kafka and produced msg using quickstart java consumer

Has it worked previously?
=========================
First attempt

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed or where adequate information has not been provided.

Please provide the relevant information for the following items:
 - [ ] SDK (include version info): Java SDK
 - [x] Sample you're having trouble with: Java quickstart
 - [x] If using Apache Kafka Java clients or a framework that uses Apache Kafka Java clients, version: 1.0.0
 - [ ] Kafka client configuration: same with sample
 - [x] Namespace and EventHub/topic name : einvoice / topic_invoice_archive
 - [x] Consumer failure
 - [ ] Timestamps *in UTC*  08:26 to 11:11 UTC (16:26 to 19:11 UTC+8)
 - [x] group.id or client.id group.id=$Default

If this is a question on basic functionality, please verify the following:
 - [x] Port 9093 should not be blocked by firewall ("broker cannot be found" errors)
 - [x] Pinging FQDN shoudl return cluster DNS resolution (e.g. `$ ping namespace.servicebus.windows.net` returns ~ `ns-eh2-prod-am3-516.cloudapp.net [13.69.64.0]`)
 - [x] Namespace should be either Standard or Dedicated tier, not Basic (TopicAuthorization errors)

**Comment by huuthanghust** [comment]
Logfile: 
First. this log show around 15 times 
INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_0, groupId=$Default] Marking the coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null) dead
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_1, groupId=$Default] Marking the coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null) dead
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_0, groupId=$Default] Marking the coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null) dead
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_1, groupId=$Default] Marking the coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null) dead
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_2, groupId=$Default] Marking the coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null) dead
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_2, groupId=$Default] Marking the coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null) dead
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_2, groupId=$Default] Discovered coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null)
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_1, groupId=$Default] Discovered coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null)
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:336: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_2, groupId=$Default] (Re-)joining group
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:336: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_1, groupId=$Default] (Re-)joining group
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_0, groupId=$Default] Discovered coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null)
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_1, groupId=$Default] Discovered coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null)
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:336: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_0, groupId=$Default] (Re-)joining group
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_ARCHIVE_10.244.4.11_0, groupId=$Default] Discovered coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null)
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:336: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_1, groupId=$Default] (Re-)joining group
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:341: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_2, groupId=$Default] Discovered coordinator einvoice.servicebus.windows.net:9093 (id: 2147483647 rack: null)
INFO   | main    | jvm 1    | 2019/09/19 19:42:30 | 19/09/2019 19:42:30  INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator.info:336: [Consumer clientId=TOPIC_INVOICE_GENERATE_PDF_INSTANCE_10.244.4.11_2, groupId=$Default] (Re-)joining group

**Comment by huuthanghust** [comment]
And then exception:
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-41" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-27" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-47" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-62" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-24" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-29" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-64" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-37" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-66" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-39" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:566)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$SyncGroupResponseHandler.handle(AbstractCoordinator.java:539)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:808)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:788)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:506)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:353)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:268)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:174)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:364)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:316)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:295)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1138)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1103)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | 	at com.viettel.einvoicegw.kafka.util.KafkaConsumer.run(KafkaConsumer.java:171)
INFO   | main    | jvm 1    | 2019/09/19 19:46:28 | Exception in thread "Thread-43" org.apache.kafka.common.KafkaException: Unexpected error from SyncGroup: The coordinator is loading and hence can't process requests.
INFO   | main

**Comment by huuthanghust** [comment]
this is KafkaConsumer code 

@Override
    public void run() {
        consumer = createConsumer();
        try {
            List<ConsumerRecord<String, String>> buffer = new ArrayList<>();
            while (true) {
                final ConsumerRecords<String, String> consumerRecords = consumer.poll(1000);
                for (ConsumerRecord<String, String> consummerRecord : consumerRecords) {
                    buffer.add(consummerRecord);
                    if (buffer.size() >= minBatchSize) {
                        logger.info("--------------------------> prepare to handleMessage");
                        handleMessage(buffer);
                        consumer.commitAsync();
                        logger.info("--------------------------> end handleMessage," + "topic: " + TOPIC + ", partition = " + consummerRecord.partition() + " lastest offset: ");
                        buffer.clear();
                    }
                }
                if(!buffer.isEmpty() && buffer != null){
                    handleMessage(buffer);
                        consumer.commitAsync();
                        logger.info("--------------------------> end handleMessage," + "topic: " + TOPIC );
                        buffer.clear();
                }
                
            }
        } catch (CommitFailedException e) {
            logger.info("CommitFailedException: " + e);
        } finally {
            consumer.close();
        }
    }

    private Consumer<String, String> createConsumer() {
        try {
            final Properties properties = new Properties();
            synchronized (KafkaConsumer.class) {
                properties.put(ConsumerConfig.CLIENT_ID_CONFIG, clientId);
            }
            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class.getName());
            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
            //Get remaining properties from config file
            properties.load(new FileReader("../conf/consumer.config"));
            consumer = new org.apache.kafka.clients.consumer.KafkaConsumer<>(properties);
            // Subscribe to the topic.
            consumer.subscribe(Collections.singletonList(TOPIC));
            return consumer;

        } catch (FileNotFoundException e) {
            logger.info("FileNoteFoundException: " + e);
            System.exit(1);
            return null;        //unreachable
        } catch (IOException e) {
            logger.info("IOException: " + e);
            System.exit(1);
            return null;        //unreachable
        }
    }

**Comment by arerlend** [comment]
I'm investigating.  Please avoid deleting the namespace if possible.

**Comment by huuthanghust** [comment]
Ok. 1 more thing. When it happened, others topics also do not consume.

**Comment by huuthanghust** [comment]
This is log that MS 've sent me 

TIMESTAMP | NamespaceName | EntityName | OperationName
-- | -- | -- | --
2019-09-06T08:21:22.5906601Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T08:22:22.6628166Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T08:23:22.6927825Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T08:24:22.6046133Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T08:25:22.6006514Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T08:26:22.6433055Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:11:22.6540016Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:12:22.7201071Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:14:22.7503368Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:15:22.7407776Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:16:22.6887938Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:17:22.7621067Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:18:22.7559601Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage
2019-09-06T11:19:22.7192510Z | eventhubskafka | topic_send_email_add_user | TryReceiveMessage

**Comment by arerlend** [comment]
Something is strange here - I've checked both 2019-9-19 11:42 UTC and 2019-9-19 19:42 UTC, and I'm not seeing any of the errors at the timestamps you provided.

Also - the last set of logs you provided are from 2019-9-06, why are these relevant?

Are the unknown errors during Sync Group ongoing issue that you are still seeing?

**Comment by huuthanghust** [comment]
1. Also - the last set of logs you provided are from 2019-9-06, why are these relevant? --> Microsoft sent me that logs (as u can see from: 8:26 -> 11:11 Kafka consumer does not send “TryReceiveMessage” to Event Hub ).
2. I still seeing unknown errors during sync group

**Comment by arerlend** [comment]
Ok let's focus on the unknown errors in sync group.  Can you give me the namespace ('eventhubskafka' or 'einvoice'), UTC timestamp, error message, and stack trace for the sync group errors?

**Comment by wely** [comment]
> Ok let's focus on the unknown errors in sync group. Can you give me the namespace ('eventhubskafka' or 'einvoice'), UTC timestamp, error message, and stack trace for the sync group errors?

@arerlend : see https://github.com/Azure/azure-event-hubs-for-kafka/issues/72#issue-496226122 all info you need are provided [here](https://github.com/Azure/azure-event-hubs-for-kafka/issues/72#issue-496226122)

**Comment by huuthanghust** [comment]
Hi @arerlend. I've provided all information you need above

**Comment by arerlend** [comment]
I believe I've figured out the cause of the unexpected sync group errors.

The session timeout set by your application is 10000 ms - however, when doing partition assignment, your sync group leader doesn't return within 10000 ms.  The service then times out everyone and returns LoadInProgress error.  The sync group leader comes back with the assignment but has already timed out, so it comes back as the only remaining member in the group.

I will look into changing the error code to something that the client expects.  However, I would recommend increasing your session timeout to at least 20000 ms, if not higher.  I don't know why the consumer is taking so long to assign but this should help mitigate the issue.

By the way, all consumers on the namespace are operating under group.id 'consumertest' - I would recommend separating applications into their own consumer groups, not just client IDs.

**Comment by huuthanghust** [comment]
Hi @arerlend, 
Thanks for your recommendations. I will increase session timeout and I also think that I should raise heartbeat.interval.ms = 6000ms (which its default is 3000ms). 
I will keep watching and I will update results here. Tks for helping.

**Comment by arerlend** [comment]
@huuthanghust let me know if this helped.  By the way, I would recommend leaving heartbeat interval at 3000 ms.  If you double your session timeout but also leave the heartbeat interval, then you're doubling the # of chances for the client to heartbeat.

**Comment by huuthanghust** [comment]
Hi @arerlend, 
It seemed to be not solution. It still happened again. 
Pls check log from your backend followed by below infos: 
EventHub/topic name : einvoice-eventhubs / topic_invoice_archive
subscriptionid: d2758da3-f3d6-417e-82bb-f08b38c7b694
Time: 11/10/2019 20:00 PM to 12/10/2019 11:30AM UTC+7

**Comment by huuthanghust** [comment]
Hi @arerlend, 
It seemed like now something happened for producer not about consumer anymore. Because I 've logged and see that the last offset is consecutive and have no problems with consumer.
 It seemed like there was some message that producer can not produce to the topic (I've try to send same messages but some of them cannot be consume).
Here is my producer.java 
    private void init() {
        try {
            logger = LoggerFactory.getLogger(KafkaProducerSingle.class);
            Properties props = new Properties();
            //thangnh config event hubs
            props.load(new FileReader("../conf/producer.conf"));

            props.put(ProducerConfig.CLIENT_ID_CONFIG, "KafkaExampleProducer");
            props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, LongSerializer.class.getName());
            props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
            logger.info("props + " + props.toString());
            //thangnh end edit
            producer = new KafkaProducer<>(props);
            producerId = System.identityHashCode(this);
        } catch (Exception e) {
            System.out.println("Failed to create producer with exception: " + e);
            System.exit(0);
        }
    }

    public void sendData(String topic, String msg) {
        try {
            if (topic != null && !topic.isEmpty()
                    && msg != null && !msg.isEmpty()) {
                producer.send(new ProducerRecord<String, String>(topic, msg));
                logger.info("KAFKA Producer " + producerId + " send to kafka topic " + topic + ", messages: " + msg);
                        }
        } catch (Exception e) {
            logger.error("KafkaProducerSingle Exception", e);
            init();
            logger.info("Retry sent data again!");
            if (topic != null && !topic.isEmpty()
                    && msg != null && !msg.isEmpty()) {
                producer.send(new ProducerRecord<String, String>(topic, msg));
                logger.info("KAFKA Producer " + producerId + " sent again when exception to kafka topic " + topic + ", messages: " + msg);
            }
        }
    }

**Comment by huuthanghust** [comment]
Hi @arerlend ,
I've enabled captured the events feature, and analyst the log and messages lost couldn't be found in there. 
Event though, I've sent same messages and application logs get succeed log

**Comment by huuthanghust** [comment]
Hi @arerlend 
I've added 
producer.send(new ProducerRecord<String, String>(topic, msg), new Callback() {
                    public void onCompletion(RecordMetadata metadata, Exception exception) {
                        if (exception instanceof NetworkException) {
                            logger.error("Exception: " + exception);
                            producer.flush();
                            producer.close();
                            init();
                        }
                    }
                });

I got the error:

ERROR com.viettel.einvoicegw.kafka.util.KafkaProducerSingle.onCompletion:76: Exception: org.apache.kafka.common.errors.NetworkException: The server disconnected before a response was received.

Anything wrong with configuration or how to handle resend messages?

**Comment by arerlend** [comment]
Sounds like the original consumer issue has been resolved.  I'll close this issue.

The producer issue is a separate issue.  I will start looking into it, but please open a new Github issue for tracking purposes with the details/logs.  NetworkException is a retriable exception, you should retry on network exception until it succeeds/until retry count is exhausted.
